src/main/java/com/shapesecurity/shift/es2017/parser/GenericParser.java

```diff
@@ -68,6 +72,26 @@ protected boolean match(@Nonnull TokenType subType) {
         return this.lookahead.type == subType;
     }
 
+    protected boolean matchIdentifier() {
+        switch(this.lookahead.type) {
+            case IDENTIFIER:
+            case LET:
+            case YIELD:
+            case ASYNC:
+                return true;
+            case AWAIT:
+                if (!this.moduleIsTheGoalSymbol) {
+                    if (this.firstAwaitLocation == null) {
+                        this.firstAwaitLocation = this.getLocation();
+                    }
+                    return true;
+                }
+                return false;
+            default:
```

Comment:
In shift-parser-js we have an [additional case](https://github.com/shapesecurity/shift-parser-js/blob/eddd5e7f0446d12460ea076b6a8fb217406eefeb/src/parser.js#L147-L156) for `ESCAPED_KEYWORD`. Is that not necessary here?

Reply:
Shift-java has no mirror of the shift-parser-js `ESCAPED_KEYWORD` case. See [issue #220](https://github.com/shapesecurity/shift-java/issues/220)